db/cache/charts.go

```diff
@@ -838,9 +842,16 @@ func (charts *ChartData) encode(sets ...lengther) ([]byte, error) {
 	if len(sets) == 0 {
 		return nil, fmt.Errorf("encode called without arguments")
 	}
+
 	smaller := sets[0].Length()
 	for _, x := range sets {
 		l := x.Length()
+
+		// Detect empty datasets and return an error if found.
+		if l == 0 {
+			return nil, fmt.Errorf("empty dataset found")
```

Comment:
I'm torn on whether this should be an error. This would not necessarily be an error for a `Days` data set on the day of a chain's genesis block, but that case is extremely limited. On the other hand, there may also be larger bins (week, month) implemented someday. 

Now that you've added the missing `Lengthen` call, all validation should already be done there, so this error is probably not needed anyway. Maybe `encode` should stay permissive of empty data sets and validation can stay in `Lengthen`.

Reply:
You are right on this that its occurrence would be rarer now but in case it ever occurred for whatever reason in the future, its effects would be disastrous. No chart that can load with one or more of the axis data set empty.
Also I don't imagine an ideal case scenario where empty data sets for the historical charts will be returned. It is not possible (that straight forward) to run the dcrdata server with only the genesis block loaded.
Still think that an error isn't suitable?

Reply:
On second though let me convert it to a warning.

Reply:
Let's get rid of it, with no warning. In the case of actual invalid data, there will already be a message from `Lengthen`. It's not necessary for `encode` to care about the content of the data.

Reply:
Just to keep things modular. Separation of concerns.